src/v/datalake/tests/record_multiplexer_bench.cc

```diff
@@ -0,0 +1,564 @@
+      std::string proto_schema,
+      std::vector<int32_t> msg_idx,
+      ::testing::protobuf_generator_config gen_config) {
+        co_await try_add_protobuf_schema(schema_name, proto_schema);
+        co_return co_await generate_batches(
+          records_per_batch, batches, [&](auto& bb) {
+              return _record_gen.add_random_protobuf_record(
+                bb, schema_name, msg_idx, std::nullopt, gen_config);
+          });
+    }
+
+    ss::future<chunked_vector<model::record_batch>> generate_avro_batches(
+      size_t records_per_batch,
+      size_t batches,
+      std::string schema_name,
+      std::string avro_schema,
+      ::testing::avro_generator_config gen_config) {
+        co_await try_add_avro_schema(schema_name, avro_schema);
+        co_return co_await generate_batches(
+          records_per_batch, batches, [&](auto& bb) {
+              return _record_gen.add_random_avro_record(
+                bb, schema_name, std::nullopt, gen_config);
+          });
+    }
+
+    datalake::record_multiplexer create_mux() {
+        return datalake::record_multiplexer(
+          ntp,
+          std::make_unique<datalake::test_serde_parquet_writer_factory>(),
+          _schema_mgr,
+          _type_resolver,
+          _translator);
+    }
+    datalake::tests::record_generator& get_record_gen() { return _record_gen; }
+
+private:
+    std::unordered_set<std::string> _added_names;
+    datalake::catalog_schema_manager _schema_mgr;
+    datalake::record_schema_resolver _type_resolver;
+    datalake::tests::record_generator _record_gen;
+    datalake::default_translator _translator;
+};
+
+PERF_TEST_CN(
+  record_multiplexer_bench_fixture, protobuf_381_byte_message_linear_1_field) {
+    static constexpr size_t records_per_batch = 10;
+    static constexpr size_t batches = 1000;
+    static constexpr ::testing::protobuf_generator_config gen_config = {
+      .string_length_range{302, 302}};
+
+    static std::string proto_schema = generate_linear_proto(1);
+
+    static thread_local chunked_vector<model::record_batch> batch_data
+      = co_await generate_protobuf_batches(
+        records_per_batch,
+        batches,
+        "proto_schema",
+        proto_schema,
+        {0},
+        gen_config);
+
+    auto reader = model::make_fragmented_memory_record_batch_reader(
+      share_batches(batch_data));
+
+    auto consumer = counting_consumer{.mux = create_mux()};
+
+    perf_tests::start_measuring_time();
+    auto res = co_await reader.consume(std::move(consumer), model::no_timeout);
+    perf_tests::stop_measuring_time();
+
+    co_return res.total_bytes;
+}
+
+PERF_TEST_CN(
+  record_multiplexer_bench_fixture,
+  protobuf_381_byte_message_linear_40_fields) {
+    static constexpr size_t records_per_batch = 10;
+    static constexpr size_t batches = 1000;
+    static constexpr ::testing::protobuf_generator_config gen_config = {
+      .string_length_range{5, 5}};
+
+    // Generate a schema near the nesting limit.
```

Comment:
nit: but there's no nesting in the linear schema, is there? Or is this referring to something else?

Reply:
Sorry, this was an artifact of copying the code from a different test. Fixing...